src/test/java/FlightsReporterTests.groovy

```diff
@@ -0,0 +1,197 @@
+import org.junit.Before
+import org.junit.Test
+import org.junit.Assert
+import org.junit.jupiter.api.BeforeAll
+
+import java.nio.channels.FileLock
+import java.time.LocalDateTime
+import java.time.LocalTime;
+//import FlightsReporter
+
+import static  org.junit.jupiter.api.Assertions.*;
+
+class FlightsReporterTests {
+    List<Flight> flights = []
+
+    @Test
+    void testFlightLengthInMinutes() {
+        Flight fl1 = new Flight(id: "SOFLON11", from: FlightsReporter.Airports.SOFIA, destination: FlightsReporter.Airports.LONDON,
+            departure: LocalDateTime.of(2021, 8, 3, 10, 30),
+                arrival: LocalDateTime.of(2021, 8, 3, 13, 15),
+                passengers: 100, capacity: FlightsReporter.CAPACITIES[3], runningLate: false)
+
+        assertEquals(FlightsReporter.flightLengthInMinutes(fl1), 165)
+
+    }
+
+    @Test
+    void testGetAllFlightsTo() {
+        FlightsReporter.printFlights(flights)
+        assert FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.AMSTERDAM)
+                == [flights[0], flights[1]]
+
+        assertEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.SOFIA),
+                Arrays.asList(flights[8], flights[9]))
+
+        assertEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.LONDON),
+                Arrays.asList(flights[2], flights[4], flights[7]))
+
+        assertNotEquals(FlightsReporter.getAllFlightsTo(flights, FlightsReporter.Airports.NEW_YORK),
+                Arrays.asList(flights[5], flights[6], flights[7]))
+    }
+
+    @Test
+    void testGetAllFlightsFrom() {
+        assert FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.SOFIA) == [flights[1], flights[2], flights[3]]
+
+        assertEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.NEW_YORK), [])
+
+        assertNotEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.BERLIN), [])
+        assertEquals(FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.PARIS),
+                [flights[0], flights[5], flights[8]])
+    }
+
+    @Test
+    void testPassengerAmount() {
+        def fl1 = FlightsReporter.getAllFlightsFrom(flights, FlightsReporter.Airports.NEW_YORK)
+        assert FlightsReporter.passengerAmount(fl1) == 0
+
+        assert FlightsReporter.passengerAmount(flights) == 1430
```

Comment:
more magic numbers. The real passengerCount is defined in another method. Not easily seen in the unit test.
ideally they would be placed closer to each other so that we really know that 1430 is correct or not.

I'd structure it 

> flight1 = createFlight(passengerCount: 5)
> flight2 = createFlight(passengerCount: 10)
> assert  FlightsReporter.passengerAmount([flight1, flight2]) == flight1.passengerCount + flight2.passengerCount

If you change to 6 - it will still work.
It will not affect another unit test (as it is data created in the test and you don't care about other data)